conformance/tests/httproute-rewrite-path.go

```diff
@@ -70,6 +70,48 @@ var HTTPRouteRewritePath = suite.ConformanceTest{
 				},
 				Backend:   "infra-backend-v1",
 				Namespace: ns,
+			}, {
+				Request: http.Request{
+					Path: "/full/rewrite-path-and-modify-headers/test",
+					Headers: map[string]string{
+						"X-Header-Remove":     "remove-val",
+						"X-Header-Add-Append": "append-val-1",
+					},
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/test",
+						Headers: map[string]string{
+							"X-Header-Add":        "header-val-1",
+							"X-Header-Add-Append": "append-val-1,header-val-2",
+							"X-Header-Set":        "set-overwrites-values",
```

Comment:
Anywhere we're asserting a set/overwrite header modification, we should maybe have that header in the original request (e.g. here add `X-Header-Set` to the original request)

Reply:
I have added `X-Header-Set` as suggested, thanks Sunjay!